[*WIP*] Implementing showcase design changes for jQuery Pt.2

amarie401 opened the pull request:
## Description

Implementing new design mockups/templates for jQuery Showcase.

@bleathem @junezhang I accidentally merged into the jquery-showcase branch on PF instead of my own jquery-showcase branch when I pushed my commit so I had to open this new PR. I will be continuing to make changes in this new PR while referencing #703 

## Current Tasks
- [X] Add set of initial components
- [X] Add get-started page
- [x] Add contribution page
- [x] Use less for styling
- [X] Implement temporary HTML + JS Linting 
- [x] Fix everything linted 
- [ ] Use one JS script file for repeating jquery code in components

## Future To-Do
- Config Travis to run .hbs to html in dist (no current rawgit link) @bleathem 
- Add any components missing or remove those that can be written in vanilla js if possible #685
- Use correct directory paths & install instructions in get-started page once pf-jquery repo is 
       complete 

## Changes

Write a list of changes the PR introduces
* Set up initial jQuery components for showcase in #692 
* Added requested changes from June in #703 
* Added get-started installation guide page
* Added contribution page

## Link to rawgit and/or image

![get-started](https://user-images.githubusercontent.com/20052391/29329543-341f186a-81c4-11e7-8f87-e4c66ea16c02.gif)

![jquqery-showcase](https://user-images.githubusercontent.com/20052391/29170432-9c24fd52-7da6-11e7-9fb7-2efabf2ee7fa.gif)
## PR checklist (if relevant)

- [ ] **Responsive**: works in extra small, small, medium and large view ports.
- [x] **Preview the PR**: An image or a URL for designer to preview this PR is provided.

amarie401 commented:
@junezhang I have added the "get-started" page. It is inside of ```tests/pages/jquery/src/get-started.html``` along with the ``` get-started.css```. Can you please review? Thank you.

junezhang commented:
@amarie401 Thanks for your PR. Some feedbacks about it as following:
- No need to add a more css file, it is same with jquery-showcase.css
- In the component page, it should have a "Get Started" link, same with "Contribute" link.
- Regs the code, the title of PATTERNFLY-JQUERY should be PatternFly jQuery, not PatternFly Showcase
`<a id="logo" title="PatternFly Showcase" href="#">PATTERNFLY-<span class="secondary-logo">JQUERY</span></a>`
- In the Get Started page, there is an introduction about PatternFly-jQuery. I'd like to update it like these:

> PaternFly-jQuery came about from our love of jQuery and PatternFly Designs. 
It will provide a set of common jQuery components for use with the PatternFly reference implementation.

How do you think about it?

amarie401 commented:
@junezhang I have made the requested changes. Please let me know if there is anything you would like to see differently. I realized that I do not have a contribute page. Should I implement the one you have already created but adapt to jQuery? Thanks!

jeff-phillips-18 commented on tests/pages/jquery/src/get-started.html:
```diff
@@ -136,7 +136,8 @@
-          This project will provide a set of common jQuery components for use with the PatternFly reference implementation.
+          PaternFly-jQuery came about from our love of jQuery and PatternFly Designs.
+          It will provide a set of common jQuery components for use with the PatternFly reference implementation.
         </p>
```
I'm not sure I like the "came about from our love" part.  

How about something like:
> Patternfly-JQuery provides a set of common components that implement PatternFly design patterns using JQuery.

amarie401 replied:
@junezhang I think we can remove the part of "came about from our love" and use what @jeff-phillips-18  has suggested

junezhang replied:
@jeff-phillips-18 @amarie401 Sure, make sense.

amarie401 replied:
@jeff-phillips-18 @junezhang it is now added

jeff-phillips-18 commented on tests/pages/jquery/src/jquery-showcase.css:
```diff
@@ -1,3 +1,5 @@
+/* CUSTOM SHOWCASE STYLES */
```
Any reason for not using less here? It would be nice to use patternfly/bootstrap variables.

amarie401 replied:
I'm glad you mentioned it. I kept going back and forth about if I should be using less or not. Thanks for the confirmation. I will add that to my current tasks.

junezhang replied:
@amarie401 the showcase less file is here: https://github.com/junezhang/showcase/blob/master/dist/less/showcase.less

amarie401 replied:
@junezhang @jeff-phillips-18 I have added the contribute page. Will work on adding the less file next.

amarie401 commented:
@junezhang @jeff-phillips-18 I have added the contribute page. Will work on adding the less file next.

amarie401 commented:
@junezhang I have added your less files. I also added your PF variable overrides to the jquery-less file directly but could not import them. I still need to make some adjustments to my custom styles for the component page to use less.

junezhang commented:
@amarie401 Thanks, I agree with you. I combined the variables.less to the showcase.less. 
https://github.com/junezhang/showcase/blob/master/dist/less/showcase.less

bleathem commented:
@amarie401 can you rebase this branch against patternfly/master?

(I'm working on updating the build script now)
